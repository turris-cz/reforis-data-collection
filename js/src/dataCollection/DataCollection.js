/*
 * Copyright (C) 2020 CZ.NIC z.s.p.o. (http://www.nic.cz/)
 *
 * This is free software, licensed under the GNU General Public License v3.
 * See /LICENSE for more information.
 */

import React, { useEffect } from "react";
import {
    API_STATE, ErrorMessage, Spinner, useAPIGet,
} from "foris";

import API_URLs from "API";
import Eula from "./Eula";

export default function DataCollection() {
    const [getSettingsState, getSettings] = useAPIGet(API_URLs.settings);

    useEffect(() => {
        getSettings();
    }, [getSettings]);

    if (getSettingsState.state === API_STATE.INIT || getSettingsState.state === API_STATE.SENDING) {
        return <Spinner />;
    }

    if (getSettingsState.state === API_STATE.ERROR) {
        return <ErrorMessage />;
    }

    return (
        <>
            <h1>{_("Data Collection")}</h1>
            <p>{_("Participate in data collection and dynamic distributed firewall.")}</p>
            <Eula eula={getSettingsState.data.eula} />
        </>
    );
}
