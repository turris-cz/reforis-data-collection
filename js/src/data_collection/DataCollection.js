/*
 * Copyright (C) 2020 CZ.NIC z.s.p.o. (http://www.nic.cz/)
 *
 * This is free software, licensed under the GNU General Public License v3.
 * See /LICENSE for more information.
 */

import React, { useEffect } from "react";

import { useAPIGet } from "foris";

import API_URLs from "API";

export default function DataCollection() {
    const [, getExample] = useAPIGet(API_URLs.example);
    useEffect(() => {
        getExample();
    }, [getExample]);

    return (
        <>
            <h1>{_("Data Collection")}</h1>
            <p>{_("Participate in data collection and dynamic distributed firewall.")}</p>
        </>
    );
}
