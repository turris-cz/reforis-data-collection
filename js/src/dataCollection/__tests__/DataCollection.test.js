/*
 * Copyright (C) 2020 CZ.NIC z.s.p.o. (http://www.nic.cz/)
 *
 * This is free software, licensed under the GNU General Public License v3.
 * See /LICENSE for more information.
 */

import React from "react";
import { render,wait } from "foris/testUtils/customTestRender";

import DataCollection from '../DataCollection';
import mockAxios from 'jest-mock-axios';


describe("<DataCollection />", () => {
    it("should render component", () => {
        const { getByText } = render(<DataCollection />);
        mockAxios.mockResponse({data: {eula: 0, token: "random_token"}});
        wait(()=>getByText("Data Collection"));
    });
});
