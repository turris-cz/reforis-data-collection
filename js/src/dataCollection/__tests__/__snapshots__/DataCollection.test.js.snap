// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DataCollection /> Should render form. 1`] = `
<DocumentFragment>
  <h1>
    License Agreement
  </h1>
  <p>
    It's required to confirm the Terms of Participation in Turris Project to participate in the threat detection program.
  </p>
  <div
    class="card p-4 col-sm-12 col-lg-12 p-0 mb-4"
  >
    <form>
      <h2>
        Terms of Participation
      </h2>
      <p>
        Your consent with the Terms of Participation allows us to analyze potential threats detected on your router and therefore to build a dynamic firewall.
      </p>
      <div
        class="form-group"
      >
        <div
          class="custom-control custom-radio"
        >
          <input
            class="custom-control-input"
            id="Agreement-0"
            name="Agreement"
            type="radio"
            value="1"
          />
          <label
            class="custom-control-label"
            for="Agreement-0"
          >
            <div>
              I accept the 
              <a
                href="#"
              >
                Terms of Participation in Turris Project (Data Collection)
              </a>
              .
            </div>
          </label>
        </div>
        <div
          class="custom-control custom-radio"
        >
          <input
            checked=""
            class="custom-control-input"
            id="Agreement-1"
            name="Agreement"
            type="radio"
            value="0"
          />
          <label
            class="custom-control-label"
            for="Agreement-1"
          >
            <div>
              I do not accept the 
              <a
                href="#"
              >
                Terms of Participation in Turris Project (Data Collection)
              </a>
              .
            </div>
          </label>
        </div>
      </div>
      <div
        class="text-right"
      >
        <button
          class="btn btn-primary  col-sm-12 col-md-3 col-lg-2"
          type="submit"
        >
          Save
        </button>
      </div>
    </form>
  </div>
  <div
    id="modal-container"
  >
    <div
      class="modal fade"
      role="dialog"
    >
      <div
        class="modal-lg modal-dialog modal-dialog-centered modal-dialog-scrollable"
        role="document"
      >
        <div
          class="modal-content"
        >
          <div
            class="modal-header"
          >
            <h5
              class="modal-title"
            >
              Terms of Participation in Turris Project (Data Collection)
            </h5>
            <button
              class="close"
              type="button"
            >
              <span
                aria-hidden="true"
              >
                Ã—
              </span>
            </button>
          </div>
          <div
            class="modal-body"
          >
            <div
              class="spinner-wrapper my-3 text-center"
            >
              <div
                class="spinner-border"
                role="status"
              >
                <span
                  class="sr-only"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`<DataCollection /> Should toggle modal. 1`] = `
"Snapshot Diff:
- First value
+ Second value

@@ -84,11 +84,11 @@
    </div>
    <div
      id=\\"modal-container\\"
    >
      <div
-       class=\\"modal fade\\"
+       class=\\"modal fade show\\"
        role=\\"dialog\\"
      >
        <div
          class=\\"modal-lg modal-dialog modal-dialog-centered modal-dialog-scrollable\\"
          role=\\"document\\"
@@ -115,22 +115,15 @@
                </span>
              </button>
            </div>
            <div
              class=\\"modal-body\\"
-           >
-             <div
-               class=\\"spinner-wrapper my-3 text-center\\"
            >
              <div
-                 class=\\"spinner-border\\"
-                 role=\\"status\\"
+               class=\\"display-linebreak text-monospace\\"
              >
-                 <span
-                   class=\\"sr-only\\"
-                 />
-               </div>
+               eula text
              </div>
            </div>
          </div>
        </div>
      </div>"
`;
